realm/src/main/java/io/realm/Realm.java

```diff
@@ -1276,6 +1276,7 @@ boolean hasChanged() {
     public void refresh() {
         checkIfValid();
         transaction.advanceRead();
+        sendNotifications();
```

Comment:
Then every thread will be notified? Which means the thread with looper will be notified more than once with nothing changed.